x-pack/solutions/security/plugins/security_solution/public/common/components/endpoint/route_capture.tsx

```diff
@@ -6,28 +6,28 @@
  */
 
 import type { PropsWithChildren } from 'react';
-import React, { memo, useEffect } from 'react';
+import React, { memo, useEffect, useMemo } from 'react';
 import { useLocation } from 'react-router-dom';
 import { useDispatch } from 'react-redux';
 import { TimelineId } from '../../../../common/types';
-import type { AppLocation } from '../../../../common/endpoint/types';
 import { timelineActions } from '../../../timelines/store';
 
 /**
  * This component should be used above all routes, but below the Provider.
  * It dispatches actions when the URL is changed.
  */
 export const RouteCapture = memo<PropsWithChildren<unknown>>(({ children }) => {
-  const location: AppLocation = useLocation();
+  const { pathname, search, hash } = useLocation();
```

Comment:
I think this change could break our functionality (endpoint management) because you are not storing the `state` ([link](https://api.reactrouter.com/v7/interfaces/react_router.Location.html)).  

We use route `state` when jumping between pages - especially when jumping to some fleet page or when coming from fleet pages to our management pages - to store data that allows us to then build "go back to" type of links. I don't think you should drop it. 

I don't remember now if we still have Redux logic around that area of building these types of URL or if we have moved away from it, but I figure I would mention it. I did a quick search inside of the `public/management` folder and I still see references to the `userChagnedUrl` redux action.

Reply:
maybe a better approach (?) would be to store everything except the `key` since that seems to be what this change is attempting to not use to trigger state changes:

```typescript
const { key: _key, ...location } - useLocation();
```

Reply:
After reading your 1st comment, but before the 2nd, completely agree, I console.log(location) and just hastily just grabbed all but key I saw there. Added it back in https://github.com/elastic/kibana/pull/217055/commits/4fe99bc093c13ae4894bfe6f971830454f9d186c . key is the only part I wanted to remove, and I see 0 references to that part of AppLocation, but I do see references to state.